src/plugins/vis_types/vega/public/components/vega_vis_component.tsx

```diff
@@ -55,7 +55,7 @@ export const VegaVisComponent = ({
     }
   }, [visData, renderComplete]);
 
-  const updateChartSize = useCallback(
+  const updateChartSize = useMemo(
```

Comment:
This is a nit, but it seems like the extra function closure is the problem, so you could keep the `useCallback`, but remove the function definition from the next link around the throttle and it should do the same thing

Reply:
this was done on purpose to avoid eslint disabling comment
![image](https://user-images.githubusercontent.com/54894989/151820784-d581686d-d385-4f82-8b0b-ffdb75aac3ad.png)